python/cudf/cudf/tests/test_pandas_comparison.py

```diff
@@ -0,0 +1,366 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+import inspect
+import textwrap
+
+import numpy as np
+import pandas as pd
+import pytest
+
+import cudf
+from cudf.testing._utils import assert_eq
+
+
+def _is_cudf(lib):
+    # TODO: Is it safe to use `if lib is cudf`? I think there are some cases
+    # where the imports of a package in different modules may not be considered
+    # identical by `is` so we have to check the name as below, but this is
+    # probably worth verifying.
+    return lib.__name__ == "cudf"
+
+
+# The parameter name used for pandas/cudf in test functions.
+_LIB_PARAM_NAME = "lib"
+
+
+def pandas_comparison_test(*args, cudf_objects=None, assert_func=assert_eq):
+    def deco(test):
+        """Run a test function with cudf and pandas and ensure equal results.
+
+        This decorator generates a new function that looks identical to the
+        decorated function but calls the original function twice, once each for
+        pandas and cudf and asserts that the two results are equal.
+        """
+        # Handle parameters (fixtures or parametrize fixtures, it doesn't
+        # matter) that are cudf objects that need to be converted to pandas.
+        nonlocal cudf_objects
+        if not cudf_objects:
+            cudf_objects = []
+        elif isinstance(cudf_objects, str):
+            cudf_objects = [cudf_objects]
+
+        parameters = inspect.signature(test).parameters
+        params_str = ", ".join(
+            f"{p}" for p in parameters if p != _LIB_PARAM_NAME
+        )
+
+        arg_str = ", ".join(
+            f"{p}={p}"
+            for p in parameters
+            if (p != _LIB_PARAM_NAME and p not in cudf_objects)
+        )
+
+        if _LIB_PARAM_NAME in parameters:
+            if arg_str:
+                arg_str += ", "
+
+            cudf_arg_str = arg_str + f"{_LIB_PARAM_NAME}=cudf"
+            pandas_arg_str = arg_str + f"{_LIB_PARAM_NAME}=pandas"
+        else:
+            cudf_arg_str = pandas_arg_str = arg_str
+
+        if cudf_objects:
+            if cudf_arg_str:
+                cudf_arg_str += ", "
+            if pandas_arg_str:
+                pandas_arg_str += ", "
+
+            cudf_arg_str += ", ".join(
+                f"{fixture}={fixture}" for fixture in cudf_objects
+            )
+            pandas_arg_str += ", ".join(
+                f"{fixture}={fixture}.to_pandas()" for fixture in cudf_objects
+            )
+
+        src = textwrap.dedent(
+            f"""
```

Comment:
You can escape the newline so there's no `\n` at the beginning (after dedent).
```suggestion
            f"""\
```